src/app/shared/services/translation.service.ts

```diff
@@ -77,11 +93,12 @@ translate(text: string) {
 
     const length = this.dictionary.value.length;
 
-    return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length) || text;
```

Comment:
@Nishanthh Any reason, you removed toLowerCase here? It is better if all the text being compared is forced into lowercase, right?

Reply:
@avikt the reason is we have a same text in lower case also in uppercase, if we convert them to lower case then we are not able to find the exact text match(case sensitive plays major role while searching the text)

Reply:
@Nishanthh Can you give me an example of this?

Reply:
@avikt we have text "CLOSED ON" also "Closed Consultations" 
"CLOSED ON" in Json it exist before the mid element of array
if we convert them to lower case it will fall on the second half and searching in the second half fails to find the text and translation is not happening regarding it

Reply:
@Nishanthh The JSON list which you are using is already sorted taking lowercases into consideration, given this, when we compare for binary search, we as well **will have to do lowercases before check**. 
You might be facing this issue because you are sorting it on your end, which **might be case sensitive**. Ideally the untouched JSON file should not cause any issues.